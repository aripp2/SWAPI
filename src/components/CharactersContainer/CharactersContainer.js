import React from 'react';
import PropTypes from 'prop-types';
import './CharactersContainer.scss';
import CharacterCard from '../CharacterCard/CharacterCard';

const CharactersContainer = ({ characters, movieInfo, updateFavs, isLoading }) => {

  const characterInfo = characters.map((character) => {
    return <CharacterCard 
            key={character.name} 
            name={character.name}
            homeworld={character.homeworld}
            population={character.population}
            species={character.species}
            films={character.films}
            updateFavs={updateFavs}  
          />
    });

  return (
     
      <section className="characterContainer_section">

        {!movieInfo &&
            <header className="favs-header">
              <h2>Your Favorite Characters</h2>
            </header>}

        {characters.length === 0 && !movieInfo && <p className="favs-prompt">Looks like you do not have any favorites yet. Head back to Movies and view their character to add some favorite charaters here!</p>}
    
        {movieInfo && <header className='scroll'>  
          <h2 className="title">{movieInfo.title}</h2>
          <div className="galaxy">
            <h3 className="away">A long time ago in a galaxy far, far away...
            </h3>
            <p>{movieInfo.opening_crawl}</p>
          </div>
        </header>}
        {isLoading && <h1>Loading ...</h1>}
        {characterInfo}
      </section>
    );
};

export default CharactersContainer;

CharactersContainer.propTypes = {
  characters: PropTypes.array.isRequired,
  movieInfo: PropTypes.any,
  updateFavs: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
};


