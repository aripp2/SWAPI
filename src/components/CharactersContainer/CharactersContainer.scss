.characterContainer_section {
  display: flex; 
  flex-wrap: wrap; 
}

