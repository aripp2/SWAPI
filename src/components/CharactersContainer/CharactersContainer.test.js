import React from 'react'; 
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import CharactersContainer from './CharactersContainer';

describe('CharactersContainer', () => {


  it('should match snapshot', () => {
    const wrapper = shallow(<CharactersContainer 
      movieInfo={[{
        characters:["https://swapi.co/api/people/2/", "https://swapi.co/api/people/3/"],
        episode_id: 1,
        opening_crawl: "Turmoil has engulfed the Galactic Republic.The taxation of trade routes to outlying star systems is in dispute.",
        release_date: "1999-05-19",
        title: "The Phantom Menace"
      }]}
      characters={[{ 
      films: ["The Empire Strikes Back", "Attack of the Clones", "The Phantom Menace", "Revenge of the Sith", "Return of the Jedi", "A New Hope"],
      homeworld: "Tatooine",
      name: "C-3PO",
      population: "200000",
      species: "Droid"
    },
    {
      films: ["The Empire Strikes Back", "Attack of the Clones", "The Phantom Menace", "Revenge of the Sith", "Return of the Jedi", "A New Hope", "The Force Awakens"],
      homeworld: "Naboo",
      name: "R2-D2",
      population: "4500000000",
      species: "Droid"
    }]} 
    updateFavs={jest.fn()}
    /> )
    expect(wrapper).toMatchSnapshot();
  });
})
