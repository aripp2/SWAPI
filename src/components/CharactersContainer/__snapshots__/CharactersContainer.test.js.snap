// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharactersContainer should match snapshot 1`] = `
<section
  className="characterContainer_section"
>
  <CharacterCard
    films={
      Array [
        "The Empire Strikes Back",
        "Attack of the Clones",
        "The Phantom Menace",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
      ]
    }
    homeworld="Tatooine"
    key="C-3PO"
    name="C-3PO"
    population="200000"
    species="Droid"
  />
  <CharacterCard
    films={
      Array [
        "The Empire Strikes Back",
        "Attack of the Clones",
        "The Phantom Menace",
        "Revenge of the Sith",
        "Return of the Jedi",
        "A New Hope",
        "The Force Awakens",
      ]
    }
    homeworld="Naboo"
    key="R2-D2"
    name="R2-D2"
    population="4500000000"
    species="Droid"
  />
</section>
`;
